backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java

```diff
@@ -2376,6 +2379,19 @@ public boolean isModularContract(@NonNull final FlatrateTermId flatrateTermId)
 		return isModularContract(ConditionsId.ofRepoId(flatrateTermRecord.getC_Flatrate_Conditions_ID()));
 	}
 
+	@Override
+	public boolean isExistsModularContract(@NonNull final IQueryFilter<I_C_Flatrate_Term> selectedContractsFilter)
+	{
+		return flatrateDAO.isExistsModularContract(selectedContractsFilter);
+	}
+
+	@Override
+	@NonNull
+	public ImmutableSet<FlatrateTermId> getModularContractIds(@NonNull final IQueryBuilder<I_C_Flatrate_Term> queryBuilder)
```

Comment:
@dm-ps  why not IQueryFilter like on the method above?

Reply:
@teosarca done in https://github.com/metasfresh/metasfresh/pull/17903